### PR TITLE
fixed: default orderby

### PR DESCRIPTION
**Pull request by taoqf**
https://github.com/typeorm/typeorm/pull/615
https://github.com/typeorm/typeorm/issues/616

### CODE REVIEW

File: test/integration/sample30-default-order-by.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import "reflect-metadata";
```
Comment (pleerock) [inline test/integration/sample30-default-order-by.ts]: Im very sorry but can you please isolate your test from samples. What you see in "integration" is old way of making tests. New way of making tests is in `functional` directory. Can you please move your test there and isolate it from sample? (e.g. create entities right in the test directory), just take a look how other tests in `functional` directory are organised. They also are testing all connection types, not only postgres

**Comment by pleerock** [comment]
what is changed in `SelectQueryBuilder.ts`? Github does not seem to show them. Can you please remove all other not necessary changes and left only what required for this PR ? (styles changes etc.)

File: src/query-builder/SelectQueryBuilder.ts [hunk 2]
```diff
@@ -1296,20 +1297,35 @@ export class SelectQueryBuilder<Entity> extends QueryBuilder<Entity> {
...
     protected createOrderByExpression() {
 
-        let orderBys = this.expressionMap.orderBys;
+        const orderBys = this.expressionMap.orderBys;
 
         // if table has a default order then apply it
         if (!Object.keys(orderBys).length && this.expressionMap.mainAlias!.hasMetadata) {
-            orderBys = this.expressionMap.mainAlias!.metadata.orderBy || {};
-        }
+            const orderBys = this.expressionMap.mainAlias!.metadata.orderBy || {};
```
Comment (taoqf) [inline src/query-builder/SelectQueryBuilder.ts]: Here is the most important change.

Comment (taoqf) [inline src/query-builder/SelectQueryBuilder.ts]: @pleerock

Comment (pleerock) [inline src/query-builder/SelectQueryBuilder.ts]: yeah I understood that. Please remove all other non-related changes from that file

File: test/integration/sample30-default-order-by.ts (hunk 1 above)
Comment (taoqf) [inline test/integration/sample30-default-order-by.ts]: after I added test for this in `functional`, the test failed. https://github.com/typeorm/typeorm/pull/621

**Comment by pleerock** [comment]
I made a fix in latest commit, please check that.

**Comment by taoqf** [comment]
Thanks.

**Pull request closed without merging**